emcc.py

```diff
@@ -2748,8 +2748,10 @@ def check_bad_eq(arg):
       settings_changes.append('PTHREADS_PROFILING=1')
       newargs[i] = ''
     elif newargs[i] == '-fno-exceptions':
+      settings_changes.append('DISABLE_EXCEPTION_CATCHING=1')
       settings_changes.append('DISABLE_EXCEPTION_THROWING=1')
     elif newargs[i] == '-fexceptions':
+      settings_changes.append('DISABLE_EXCEPTION_CATCHING=0')
```

Comment:
I seem to remember this was deliberate.  @juj would have more info.

Reply:
My recollection was that `DISABLE_EXCEPTION_THROWING` was added to get the most out of code size (disallow even throwing). Which does not conflict with this.

Overall, if the user says `-fexceptions` or the reverse, I think we should follow that and enable (or the reverse) exceptions entirely.

Reply:
Makes sense to me.  Can we wait for @juj to weigh in?

Reply:
Certainly, I didn't mean to imply there was a rush.

Reply:
@sbc100 been a week, what do you think about landing this?

Reply:
sgtm, we can revert if @juj comes back with more info there.